packages/flutter/lib/src/rendering/box.dart

```diff
@@ -583,12 +585,15 @@ class BoxConstraints extends Constraints {
       return true;
     if (runtimeType != other.runtimeType)
       return false;
-    final BoxConstraints typedOther = other;
-    assert(typedOther.debugAssertIsValid());
-    return minWidth == typedOther.minWidth &&
-           maxWidth == typedOther.maxWidth &&
-           minHeight == typedOther.minHeight &&
-           maxHeight == typedOther.maxHeight;
+    assert(() {
```

Comment:
I believe the code here would be clearer if this is expressed as:
```dart
final BoxConstraints typedOther = other as BoxConstraints;
assert(typedOther.debugAssertIsValid());
return minWidth == typedOther.minWidth && ...
```

We know after the `if (runtimeType != other.runtimeType)` that it has to be castable to BoxConstraints, so there shouldn't be a need to check that again?
```

Reply:
I go this way:
* to keep the pattern of _is-check_ on `other` to promote its type and avoid introducing another variable.
* to avoid triggering the lint [test_types_in_equals](https://dart-lang.github.io/linter/lints/test_types_in_equals.html) when `as` is used without `is Xxx`

>We know after the `if (runtimeType != other.runtimeType)`

Unfortunately the type system don't. [There is currently no way to derive type promotion from `runtimeType` checks](https://github.com/dart-lang/sdk/issues/35524#issuecomment-467350674)

Reply:
Thanks for sharing your reasoning. It's unfortunate that test_types_in_equals can't be made smarter because of the runtimeType check constraint. I guess this is the best we can do then. (Sorry for leaving this comment everywhere on the other PRs as well)

Reply:
>(Sorry for leaving this comment everywhere on the other PRs as well)

Should I put a comment on all PR with only the same kind of comments to request a LGTM?

Reply:
No need to. I'll re-review them in a bit.